src/lib.rs

```diff
@@ -17,7 +17,9 @@ mod compat;
 mod connect;
 mod handshake;
 #[cfg(feature = "stream")]
-pub mod stream;
+pub(crate) mod stream;
```

Comment:
nit: err, does the `pub(crate)` make any difference for these top-level items?

Reply:
No differences there. I did this together with the connect/tls split to have a nicer public API.
The only thing that's available from the stream module is the `MaybeTlsStream` and that's now re-exported on the top-level for a simpler API structure.

I can of course revert this back if you prefer to have this module public and the `MaybeTlsStream` not at the top-level.

Reply:
Oh, I expressed myself to vaguely. I like `MaybeTlsStream` being promoted to the top-level.

I meant this simple change (without any intended change of functionality):
```suggestion
mod stream;
```
That seems to work? Same for `mod tls`.